### DISCUSSION TITLE
NBMiner v40.0

### DISCUSSION THREAD

**Discussion by nebutech-admin** [NebuTech/NBMiner#651] [General]
- `feature`: Add overclock options for Nvidia GPUs, require admin priviledge, accept comma seprated list to specify for each GPU.

  - `-power-limit, -pl`: Set power limitation of GPU. Examples:

    ​								  Set PL in watts: `-pl 200`

    ​                                  Set PL in percentage of default PowerLimit: `-pl 75%` (in Windows bat file, need dual `%` , `-pl 75%%`)

  - `-cclock`: Set core clock in MHz. Examples:

    ​		        Set clock offsets: `-cclock 100`  (Windows only)

    ​				Set locked clocks: `-cclock @1500`

  - `-mclock`: Set memory clock offsets in MHz (Windows only)

  - `-lock-cv`: Set locked core voltage of GPU in mV, support Turing and newer GPUs. (Windows only)

  - `-fan`: Set fan speed in percentage of GPU. (Windows only)

- `feature`: Display current LHR value in console summary table and web monitor.

- `feature`: `-proxy` options now support username & password for SOCKS5 proxy. format: `-proxy user:pass@host:port`

- `feature`: Web monitor changes, delete unused information for dual mining.

- `feature`: new options for log control.

  - `-log-no-job`: Turn off the `New job` line in console.
  - `-log-cycle`: Set to change the cycle of Summary table show in console and log, in seconds, defaults to 30.

- `fix`: `ethash` Change CPU share validation to independent thread, lower performance degradation when mining at very low difficulty pools. 

- `fix`: LHR lock detected failure on some cases.

<hr /><em>This discussion was created from the release <a href='https://github.com/NebuTech/NBMiner/releases/tag/v40.0'>NBMiner v40.0</a>.</em>

**Comment by Ildar2015** [comment]
RX6600 dont working ETH. win10 64. driver 21.10.2

**Reply by ginwer** [reply]
How to choose GPUS for diffenent set

**Reply by liukaixin007** [reply]
After using the -proxy parameter, do YOU still need the -o pool option? I don't really understand, can you give me an example?

**Comment by hendra2222** [comment]
using stratum generator with nicehash this version does not work when mining eth. rolling back to version NBMiner_39.7_Win

**Reply by damjanmiklos** [reply]
it works for me

**Comment by georgesimbu** [comment]
It's working for me

**Comment by Faltskog** [comment]
The "start_eth_overclock.bat" works with 3080ti? because in MSi Afterburner i can only hit PL 82%

**Comment by Underc0vernerd** [comment]
Since the update i cannot get my miner to go to LHR mode 1. Tried all the commands and nothing works. It starts up every time telling me im in LHR mode 2 - 71.

**Reply by FrancisGuindon** [reply]
You need to use double dash ("--") to make it work. At least, for NiceHash.

**Reply by tattooseanohara** [reply]
I use -lhr-mode 1, with no issues.  Running nbminer stand alone, but thru the nicehash pool.  I find that if I run the latest versions thru the nicehash app, I get errors.  Even though you you are able to update the nbminer version yourself in nicehash, I've found that only the version they provide works properly.  I just write the bat file to mine thru nicehash, and I am able to collect my earnings thru the NH app

**Comment by iRuLLezZ** [comment]
What is modify_tdr_delay?

**Reply by cowrevenge** [reply]
Does this cause blue screen errors? I get a random NVidia error that resets the pc. I might be 4 days, it might be 2 hours. Its not always the same GPU either, as I can see the fans shut down before the reset occurs.

**Comment by blackb1rd01** [comment]
Usando nicehash para eth el comando lhr 1 no funciona con un rtx 3060 lhr v2. Se bloquea y al aplicar el desbloqueo falla y el hashrate cae hasta 0.5 mhs
Probé volviendo a la versión 39.7 y el problema desapareció

**Comment by m1k3b13n** [comment]
no matter what I do (windows), even running normal bat without overclocks, same setup as 39.7, this version is constantly crashing at startup. starts to generate DAGS and crash. idk if maybe it's not liking my mix of nvidia and amd, but never had this problem before. errors arent specifying a device error, just process crash. had to roll back, really wanted to ditch AB again bc this is the only miner that recognizes ga104 as lhr and supports amd

**Reply by nebutech-admin** [reply]
can you post a screenshot about the crash?

**Comment by m1k3b13n** [comment]
absolutely:
![20211123_215423](https://user-images.githubusercontent.com/94943112/143163168-01549d88-65ba-464c-8bb4-61027e5c6e42.jpg)

**Reply by nebutech-admin** [reply]
@m1k3b13n which GPUs are you using? Try use only nvidia GPU by add `-platform 1`

**Comment by nebutech-admin** [comment]
@m1k3b13n which GPUs are you using? Try use only nvidia GPU by add `-platform 1`

**Reply by m1k3b13n** [reply]
runs on platform 1, does NOT run on platform 2 and both platforms simultaneously. besides the 5 nvidias (3060, 3060, 3070, 2080super, 1660ti) I have 2 amd (6600, 6700xt). specifying platform 2 crashes. so it's the amd that's crashing it

**Reply by nebutech-admin** [reply]
I'll fix a binary soon

**Reply by nebutech-admin** [reply]
@m1k3b13n check the updated binary files.

**Comment by tattooseanohara** [comment]
Are you able use negative memory overock settings, ie -502? Or will the "-" confuse the code?
Crashed on me trying set overclocks in the bat file (eth), but works fine without.  Checked how the code overclocked my cards(nvidia 30s) in afterburner, and they were all +300, not -502, after crash.  Will have to do a bit more debugging of the issues, but wanted a little more clarity first.  Very pleased to have these features in the works, thanks guys!

**Comment by FrancisGuindon** [comment]
"-log-no-job -log-cycle 15" (I've tried double dash as well) both commands don't seem to work on Windows, using NiceHash. I know it's not fully supported, but I've specified the args inside the benchmark tab.
Is there a log file I can see with all the arguments used? The CMD doesn't show what args are being used...

**Comment by wmm1337** [comment]
hi, i have a question. start_eth_overclock does the file fit the gpu setting into the windows or when opening bat file??

**Comment by jykelone** [comment]
i have six  gup, how setting ?

**Reply by jykelone** [reply]
t-rex?

**Comment by m1k3b13n** [comment]
similar to trex separations. use , in between each value, no space. use the
@ in cclock to lock core, otherwise + or -- for cclock and mclock

**Reply by JamieTx** [reply]
nbminer -a ethash -o stratum+tcp://asia2.ethermine.org:14444 -u 0x4296116d44a4a7259B52B1A756e19083e675062A.default -log -pl 70%% -cclock @1300 -mclock 1200 -fan 90, -pl 70%% -cclock @1400 -mclock 1000 -fan 100
I use the above parameter configuration and want to realize the independent configuration of overclocking for each GPU, but NBMiner will configure "-pl 70%% -cclock @1400 -mclock 1000 -fan 100" to two graphics cards at the same time. Is it my misunderstanding that causes the configuration error?

ref：feature: Add overclock options for Nvidia GPUs, require admin priviledge, accept comma seprated list to specify for each GPU.

**Reply by JamieTx** [reply]
Ok, I misunderstood, the correct command format should be:
nbminer -a ethash -o stratum+tcp://asia2.ethermine.org:14444 -u 0x4296116d44a4a7259B52B1A756e19083e675062A.default -log -pl 70%% -cclock @1300,@1400 -mclock 1200,1000 -fan 90,100

**Comment by vlknkzl** [comment]
Hello, how can I run nb miner with rx580 video card. I downloaded the winzip file but I don't know how to run it can you help me

**Comment by Mrly12655775** [comment]
超频后退出程序先前的超频设置不会恢复默认，希望能够更改

**Comment by niazmmorshed** [comment]
How can i run AMD and nvidia both but I would like to use locked core clock for nvidia. I am using following parameter for nvidia but If i add amd in the bat file, how do i do it?

@cd /d "%~dp0"
nbminer -d 0,1,3,5 -a ethash -o stratum+tcp://us1.ethermine.org:4444 -u XXXXXXXXXXXXXXXXXXX -log --lhr 74,0,74,74 -pl 63%%,63%%,65%%,55%% -cclock @975,@1215,@1065,@1035 -mclock 1600,1200,1200,1800 -fan 75,100,100,75
pause

**Comment by YuFeng-ovo** [comment]
-power-limit, -pl: Set power limitation of GPU

why it cant work，i use this
nbminer -a ethash -o stratum+tcp://eth.ss.poolin.me:443 -pl 75%% -u xxx -log
But my GPU usage is still 100%

Can anyone help me, much appreciated!

**Reply by YuFeng-ovo** [reply]
i try "--pl 35%%" and "--power-limit" still cant work

**Reply by YuFeng-ovo** [reply]
By the way
I hope that the GPU usage can be reduced to the level I specified, such as 35%
I can use the remaining usage to play games

but no matter what the settings are, the usage still reaches 100%.

If there is any other way to achieve this, please help me
Grateful!
Grateful!

**Reply by hendra2222** [reply]
I Have 3060 and I play games while I mine all the time. No impact no problem. Even the GPU usage is 100%, when you play games it prioritized on games more than mining (notice mining drop when playing games)..

**Reply by YuFeng-ovo** [reply]
> I Have 3060 and I play games while I mine all the time. No impact no problem. Even the GPU usage is 100%, when you play games it prioritized on games more than mining (notice mining drop when playing games)..

Does this mean that the functionality I want cannot be achieved.
What function is the --pl parameter actually used for?

(Although the problem is solved, I still want to know)

Thanks for your answer to me!

**Reply by hendra2222** [reply]
dont know.. I never used --pl. everything's default

**Reply by Tom5051** [reply]
-pl limits the GPUs maximum wattage to whatever you set it to. Fairly certain this would affect gaming FPS as it is a GPU wide wattage limit.
I have LHR cards and use -pl 140 for 3060 cards, -pl 285 for 3080tis and -pl 130 for 1070s. for multiple cards it is set -pl 140,140,140,140,285,285,130,130,130  for example

**Reply by hendra2222** [reply]
cool ohnestly never used it. I have After burner to take care of it. currently my 3060 is on 101w, never had issue with games all runs smooth while gaming even the most intensive cyberpunk. I am on RYZEN 5 5600X incase you guys interested

**Reply by YuFeng-ovo** [reply]
Thanks a lot for everyone's answers!
I think I understand now

-pl just limits the power of the GPU
But this will not necessarily reflect the value seen on the Windows Task Manager
Let the task manager mislead before (it's also possible that -pl didn't really work)

But in any case, the impact on my game should not be very big
